src/tests/domain/provider-sessions.spec.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2019 The "mysteriumnetwork/mysterium-vpn-js" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import lolex, { InstalledClock, NodeClock } from 'lolex'
+import { ServiceSessionDTO } from 'mysterium-tequilapi/lib/dto/service-session'
+import { ProviderSessions } from '../../domain/provider-sessions'
+import { EmptyTequilapiClientMock } from '../utils/empty-tequilapi-client-mock'
+import { nextTick } from '../utils/utils'
+
+class ProviderServiceTequilapiClientMock extends EmptyTequilapiClientMock {
+  public serviceSessionsMock: ServiceSessionDTO[] = [
+    { id: 'id1', consumerId: '0x1' }
+  ]
+
+  public async serviceSessions (): Promise<ServiceSessionDTO[]> {
+    return Promise.resolve(this.serviceSessionsMock)
+  }
+}
+
+describe('ProviderSessions', () => {
+  let service: ProviderSessions
+  let tequilapiClient: ProviderServiceTequilapiClientMock
+  let clock: InstalledClock<NodeClock>
+
+  beforeAll(() => {
+    clock = lolex.install()
+  })
+
+  afterAll(() => {
+    clock.uninstall()
+  })
+
+  beforeEach(() => {
+    tequilapiClient = new ProviderServiceTequilapiClientMock()
+    service = new ProviderSessions(tequilapiClient)
+  })
+
+  describe('.onCount', () => {
+    it('doest not invoke callback initially', async () => {
```

Comment:
If this does not notify initially, how to get initial value? When switching between VPN and provider tabs, this component will be rerenderer - then we might need to get initial value.

Reply:
Where is no initial value is `undefined` until it will be notified by subscription